raven/utilities/gis.py

```diff
@@ -99,3 +99,87 @@ def aggfunc(x):
             return x[0]
 
     return gdf.dissolve(by='MAIN_BAS', aggfunc=aggfunc)
+
+
+def get_bbox(fn):
+    """Return bounding box of first feature in file.
+
+    Parameters
+    ----------
+    fn : str
+      Path to file storing vector features.
+
+    Returns
+    -------
+    list
+      Geographic coordinates of the bounding box (lon0, lat0, lon1, lat1).
+
+    """
+    for i, layer_name in enumerate(fiona.listlayers(fn)):
+        with fiona.open(fn, 'r', layer=i) as src:
+            for feature in src:
+                geom = shape(feature['geometry'])
+                return geom.bounds
+
+    # for i, layer_name in enumerate(fiona.listlayers(fn)):
+    #     with fiona.open(fn, 'r', layer=i) as src:
+    #         return src.bounds
+
+
+def get_dem(bbox):
+    """
+    Return a subset of the EarthEnv NorthAmerica DEM.
+
+    Parameters
+    ----------
+    bbox : sequence
+      Geographic coordinates of the bounding box (lon0, lat0, lon1, lat1).
+
+    Returns
+    -------
+    bytes
+      A GeoTIFF array.
+
+    """
+    # size: 143999, 87599
+    # bbox: -170, 10, -50, 83
+    # dlon = dlat = 1200
+    from owslib.wcs import WebCoverageService
+    from lxml import etree
+
+    # We should use 2.0.1. It is supported by GeoServer and the previous versions are deprecated.
+    version = '1.0.0'
+    crs = 'epsg:4326'
+    fmt = 'GeoTIFF'
+    (lon0, lat0, lon1, lat1) = bbox
+
+    wcs = WebCoverageService('http://boreas.ouranos.ca/geoserver/ows', version=version)
+
+    # Need to compute the width and height based on the resolution.
+    if version == '1.0.0':
+        layer = 'public:EarthEnv_DEM90_NorthAmerica'
+        resp = wcs.getCoverage(identifier=layer,
+                               bbox=bbox,
+                               format=fmt, crs=crs,
+                               width=int((lon1 - lon0) * 10), height=int((lat1 - lat0) * 10))
+
+    # This is not working.
+    elif version == '2.0.1':
+        layer = "public__EarthEnv_DEM90_NorthAmerica"
```

Comment:
This was changed from `public:EarthEnv_DEM90_NorthAmerica` based on what I saw in the WCS GetCapabilities xml

Reply:
Yep, not sure why.

Reply:
Yeah, me neither. If it works fine the previous way, I'll change it back. I may have jumped the gun on this one.